### DISCUSSION TITLE
ReferenceError: Cannot access 'Entity' before initialization

### DISCUSSION THREAD

**Discussion by thanhtung0212** [vndevteam/nestjs-boilerplate#218] [Q&A] [answered]
hi @khuongln-1346 . i have some problem when create new Entity. I have created 2 entities and import parallel. 
but it show error message like this: 
`ReferenceError: Cannot access 'BrandEntity' before initialization`
Can you help me resolve it or give me solution for this? thanks you 
cc @xxiihdc 

![Screenshot From 2024-08-22 17-01-19](https://github.com/user-attachments/assets/b654afff-a79e-47e5-97d5-bc23d6e29dcd)

**Comment by khuongln-1346** [comment]
This boilerplate is using `SWC` compiler
So, you can refer to [this doc from nestjs](https://docs.nestjs.com/recipes/swc#common-pitfalls) to resolve this problem

Ex:

```
@Entity()
export class User {
  @OneToOne(() => Profile, (profile) => profile.user)
  profile: Relation<Profile>; // <--- see "Relation<>" type here instead of just "Profile"
}
```

**Reply by thanhtung0212** [reply]
thanks @khuongln-1346 . Looks like it's still errors in DTO folder when i imported it. maybe it doesn't work in DTO 
`ReferenceError: Cannot access 'BrandManagersResDto' before initialization`
![Screenshot From 2024-08-22 21-44-27](https://github.com/user-attachments/assets/aa6d03b5-002a-44e4-a061-5a51b3ed9c03)

**Reply by lamngockhuong** [reply, marked as answer]
@thanhtung0212 This is a bug 🙇 
I fixed it https://github.com/vndevteam/nestjs-boilerplate/pull/220

**Reply by thanhtung0212** [reply]
Thanks you so much. That's work well for me
